src/main/webapp/scripts/sessioncache.js

```diff
@@ -38,36 +44,48 @@ class SessionCache {
      */
     this.refreshCadence_ = refreshCadence;
 
+    /**
+     * Represents the handler returned by the setTimeout that refreshes.
+     * @private {number}
+     */
+    this.setTimeoutId_ = 0;
+
     /**
      * @private {Object}
      */
     this.urlParams_ = urlParams;
   }
 
   /**
-   * Refreshes the result from the session information poller 
+   * Refreshes (rate dictated by the refreshCadence) 
+   * the result from the session information poller 
    * and updates the sessionInformation field.
-   * Keys are updated every 30 seconds.
    * @private
    */
-  refreshSessionInformation_() {
-    throw new Error('Unimplemented');
+  async refreshSessionInformation_() {
+    this.sessionInformation_ = 
+        await this.sessionInformationPoller_.getLastResult();
+    this.setTimeoutId = setTimeout(() => {
+      this.refreshSessionInformation_();
+    }, this.refreshCadence_);
   }
 
   /** 
    * This method begins polling for session information and starts
    * refreshing.
    */
   start() {
-    throw new Error('Unimplemented');
+    this.sessionInformationPoller_.start();
+    this.refreshSessionInformation_();
```

Comment:
I dont understand this code. The poller already polls, doesnt it? Why does the refresh cache need to have a setTimeout call of its own? All the cache has to do is start the poller at the proper cadence and that's it. no?

Reply:
Is refresh not meant to periodically obtain the results from the poller or should that just be done at getSessionInformation() and get rid of refreshSessionInformation_